text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,416 @@
+    team has, as it enables our progress in the compiler's continued development
+    and maintenance.
+- **organization membership**
+  - Contributors that are added to the *rust-lang/compiler* team in the GitHub
+    organization can be assigned to issues/pull requests, modify labels, receive
+    group mentions and receive a "Member" badge next to their name.
+- **rustc-perf**
+  - Contributors with permissions to use *rustc-perf* can request benchmarking
+    of their pull requests (and pull requests they are reviewing). *rustc-perf*
+    permissions are useful for regular contributors as it is common to need to
+    request benchmarks from contributors with permissions. *rustc-perf* permissions
+    only make sense alongside *try* permissions.
+- **crater**
+  - Contributors with permissions to *crater* can request crater runs to check
+    whether their code breaks any public ecosystem code. *crater* permissions only
+    make sense alongside *try* permissions.
+- **dev desktops**
+  - Contributors with access to developer desktops are able to connect to shared
+    development servers that they can do their contributions from.
+- **triagebot**
+  - [triagebot][triagebot] is a GitHub bot that can perform helpful tasks on issues
+    and pull requests. Many of its functions are available to everyone, such as issue
+    claiming, but some functions may be restricted to project/team members.
+
+[expectations]: https://forge.rust-lang.org/compiler/reviews.html#expectations-for-r
+[triagebot]: https://forge.rust-lang.org/triagebot/index.html
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Contributors start without any particular privileges, permissions or
+responsibilities and can contribute whatever they'd like. Contributors can
+progress to [Team Members][team-members] and then [Maintainers][maintainers].
+
+## Team Members
+[team-members]: #team-members
+
+Being able to grant permissions to trusted contributors quickly is beneficial to
+enable them to contribute to the project more efficiently and review and approve
+work of their collaborators.
+
+Any contributor can request to become a team members by contacting the
+compiler team's leads, or current maintainers and team members can
+nominate a contributor. Team leads will check for a reasonable contribution
+history, and will check if the current team have any serious concerns related to
+contributor conduct (waiting approximately one week).
+
+When evaluating a candidate's contribution history, length of time and
+consistency of contributions and interactions with other contributors and
+maintainers will be taken into account. It is important to note that many kinds
+of contributions will be considered such as code contributions, helping with
+issue triage and bisection, running meetings and creating minutes, documentation
+contributions for rustc internals or the [Compiler Development Guide], etc.
```

Comment:
I understand that it is possible to easily attach an objective criteria to the code contributions (number of merged pull requests, size of them) after the fact. What tools does GitHub (or our infrastructure in general) provide to list actions pertaining to triage, bisection, etc? Do we anticipate those to remain an action that a current member would need to consciously remind themselves to notice?

Reply:
I can probably partially answer to this.

Non-code contributions to the project are basically invisible i.e. hard to quantify and produce stats for (unless, well, you look at the # of comments on GH and Zulip but they give a partial view[^1]). We have tooling to facilitate this work (f.e. triagebot) and people like me are working to improve this tooling.

> Do we anticipate those to remain an action that a current member would need to consciously remind themselves to notice?

IIUC your question you're asking if people doing non-code work know what to do and where to put their fingers on. The answer is yes and no because organizational work is relatively non-structured but we are looking into improving this. 

---

[^1]: reminds me when Apple used to measure developer throughput by counting LOCs/week ([link](https://computerhistory.org/blog/macpaint-and-quickdraw-source-code/?key=macpaint-and-quickdraw-source-code))